src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -500,13 +499,21 @@ class DevTask extends AbstractFeatureTask {
         @Override
         public boolean updateArtifactPaths(ProjectModule projectModule, boolean redeployCheck, boolean generateFeatures, ThreadPoolExecutor executor)
                 throws PluginExecutionException {
-            // not supported for Gradle, only used for multi module Maven projects
+            // unable to identify the changes made, showing option for user. always return false because action is invoked manually
+            if (isMultiModuleProject()) {
+                info("A change was detected in a build file. The libertyDev task could not determine if a server restart is required.");
+                info("To restart the server, type 'r' and press Enter.");
+            }
             return false;
```

Comment:
So by returning false in these two `updateArtifactPaths` nothing will be done (as your comment says). Should we return true instead so that the recompile is done? We still wouldn't know if a restart of the server was necessary, but at least we would take care of the case where it was a dependency change. And I think Gradle may optimize itself for the compile if there was no change that affected the compile (but I am not sure). What do you think?

Reply:
We can do that, we can set updateArtifacts to return true and recompileDependencies as true, this will trigger recompiling.. but due to the recompile related messages, the message we specified above will not be visible in the expected place. do you want to make the changes

Reply:
Ok, so what if we don't even call updateArtifactPaths if it is a multimodule project? In the two places updateArtifactPaths is called, we can check first if it is multimodule project. If it is, due the recompile and then print the message. What do you think?